### PR TITLE
Studio: Fix the error when Studio cannot update thumbnail for a stopped server

### PR DESCRIPTION
**Pull request by katinthehatsite**
## Related issues

Closes https://github.com/Automattic/dotcom-forge/issues/9701

## Proposed Changes

This PR ensures that we do not attempt to update a thumbnail when the server is stopped and we skip this step. It will also clean up the Sentry logs for `Studio cannot update thumbnail for a stopped server`.

From digging in https://github.com/Automattic/dotcom-forge/issues/9701 , I discovered that the issue usually happens in relation to the `Out of memory` error related to PHP WASM in the `startServer` process and mostly on Windows machines. This can be confirmed by checking Sentry logs ([there are some linked in this comment](https://github.com/Automattic/dotcom-forge/issues/9701#issuecomment-2473643204)). It seems that the scenario where this error happens works like this:

* The server starts and `running` is set to `true`
* The process proceeds with the update of the thumbnail
* In the process of attempting the thumbnail, the server crashes and running is set to `false`
* The thumbnail update fails and throws the error `Studio cannot update thumbnail for a stopped server` logged in Sentry

I propose the following:

* Adding a check if the server is running right before attempting to call `updateCachedThumbnail`.  In that case, if the server is not running, it won’t even attempt the `updateCachedThumbnail call`, saving processing time and avoiding unnecessary logging. 
* Additionally, even though `updateCachedThumbnail` has its own safety checks, ensuring the precondition here makes the calling code more robust and explicitly documents that the thumbnail should not be updated if the server is not running. 

## Testing Instructions

There is no easy way to test this. I tried triggering this error on macOs and was not able to. For now, start Studio with `npm start` and confirm that the thumbnail still gets updated correctly when you start and stop a site:

* Pull the changes from this branch
* Start Studio with `npm start`
* Start a local Studio site and confirm that the thumbnail looks good
* Open wp-admin of the local Studio site and Studio side by side
* Change a theme on a local Studio site in wp-admin and super quickly stop the server in Studio before the thumbnail has a chance to update
* Start the site and confirm the thumbnail updates

## Pre-merge Checklist

- [ ] Have you checked for TypeScript, React or other console errors?

### CODE REVIEW

**Comment by katinthehatsite** [comment]
I am wondering if we might want to wrap:

```
await server.start();
```

in the [try/catch block and return early](https://github.com/Automattic/studio/blob/10f7571167bdc8119972a1530226019b79ba77ef/src/ipc-handlers.ts#L311) from the function or if that would be excessive.

File: src/ipc-handlers.ts [hunk 1]
```diff
@@ -342,7 +342,17 @@ export async function startServer(
...
+
+	if ( server.details.running ) {
+		server
+			.updateCachedThumbnail()
+			.then( () => {
+				sendThumbnailChangedEvent( event, id );
+			} )
+			.catch( ( error ) => {
+				console.error( `Failed to update thumbnail for server ${ id }:`, error );
+			} );
```
Comment (ivan-ottinger) [inline src/ipc-handlers.ts]: Maybe we could consider using try/catch for better readability:

```suggestion
		try {
			await server.updateCachedThumbnail();
			sendThumbnailChangedEvent( event, id );
		} catch ( error ) {
			console.error( `Failed to update thumbnail for server ${ id }:`, error );
		}
```

**Review by ivan-ottinger — APPROVED** [review]
The proposed solution looks good to me. 🙂👍🏼 

I did not observe any issue in my tests. Sharing just one idea in the inline comment below.

**Comment by katinthehatsite** [comment]
Sounds good, thanks @ivan-ottinger ! I applied your suggestion!

**Pull request merged**